Nodejs/Product/Nodejs/NpmUI/NpmWorker.cs

```diff
@@ -377,6 +379,67 @@ public void Dispose()
             this.commandQueue.CompleteAdding();
         }
 
+        // TODO: This is duplicated from Npm project
+        // We should integrate this into the NpmCommander
+        internal static async Task<IEnumerable<string>> ExecuteNpmCommandAsync(
+            string pathToNpm,
+            string executionDirectory,
+            string[] arguments,
+            bool visible = false,
+            Redirector redirector = null)
+        {
+            IEnumerable<string> standardOutputLines = null;
+
+            using (var process = ProcessOutput.Run(
+                pathToNpm,
+                arguments,
+                executionDirectory,
+                /*env*/ null,
+                visible,
+                redirector,
+                quoteArgs: false,
+                outputEncoding: redirector == null ? null : Encoding.UTF8
+                ))
+            {
+                var whnd = process.WaitHandle;
+                if (whnd == null)
+                {
+                    // Process failed to start, and any exception message has
+                    // already been sent through the redirector
+                    if (redirector != null)
+                    {
+                        redirector.WriteErrorLine("Error - cannot start npm");
+                    }
+                }
+                else
+                {
+                    var i = await Task.Run(() => WaitHandle.WaitAny(new[] { whnd }));
+                    if (i == 0)
+                    {
+                        Debug.Assert(process.ExitCode.HasValue, "npm process has not really exited");
+                        process.Wait();
+                        if (process.StandardOutputLines != null)
```

Comment:
>StandardOutputLines [](start = 36, length = 19)

I've had problems with this method of redirecting output.  In particular, if the output buffer fills up, the process may pause until something consumes it (which won't happen until it completes, which won't happen until something consumes the output, ...).  I have a sample implementation of a different pattern, if you're interested.

Reply:
wrapper does the right thing